packages/mobile/src/web3/saga.ts

```diff
@@ -424,8 +429,20 @@ export function* switchToGethFromZeroSync() {
   try {
     const gethAlreadyStartedThisSession = yield select(gethStartedThisSessionSelector)
     yield put(setZeroSyncMode(false))
-    yield call(initGethSaga)
 
+    if (gethAlreadyStartedThisSession) {
+      // Call any method on web3 to avoid a persist state issue
+      // This is a temporary workaround as this restart will be
+      // removed when the geth issue is resolved
+      yield call(web3.eth.isSyncing)
+      // If geth is started twice within the same session,
+      // there is an issue where it cannot find deployed contracts.
+      // Restarting the app fixes this issue.
+      restartApp()
```

Comment:
Should we communicate this to the user somehow?

Reply:
It's communicated in a modal in CeloLite that pops up if a restart will be needed